### PR TITLE
Prepare for exception classes in identifyException

### LINKED ISSUE #15965: CL22027 integration points for heritage JDBC behavior
User story:
"As developers, we need a way to achieve compatibility with various legacy JDBC behaviors so that apps which depend on these behaviors can run."

### PR DESCRIPTION
**Pull request by KyleAure**
PR opened on behalf of @njr-11.

### CODE REVIEW

File: dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/DataSourceService.java [hunk 1]
```diff
@@ -849,21 +850,45 @@ protected void modified(ComponentContext context, Map<String, Object> newPropert
...
+            int dot2 = key.length();
+            key.append(".as");
+            String as = (String) configProps.get(key.toString());
+            if (as == null)
+                break; // no more nested identifyException elements
+            key.replace(dot2 + 1, dot2 + 9, "sqlState");
+            String sqlState = (String) configProps.get(key.toString());
+            key.replace(dot2 + 1, dot2 + 10, "errorCode");
+            Integer errorCode = (Integer) configProps.get(key.toString());
+            key.delete(18, key.length());
```
Comment (KyleAure) [inline dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/DataSourceService.java:875]: This all seems like unnecessary manual string manipulation. 
Couldn't this be done by using a formatted string:
```java
String keyFormat = "identifyException.%d.%s"
String key;
for (int i = 0; i < 1000; i++) { // cardinality is capped at 1000 in metatype
  key = String.format(keyFormat, i, "as");
  ...
  key = String.format(keyFormat, i, "sqlState);
  ...
}
```

Would make things more readable, but I'm not sure the performance difference between a StringBuilder and StringFormatter.

**Review by njr-11 — APPROVED** [review]

**Comment by njr-11** [comment]
#run-libby-bot

**Comment by njr-11** [comment]
The changes that Kyle requested were added in the above commit

**Comment by njr-11** [comment]
#build

**Pull request merged**